product.py

```diff
@@ -280,6 +280,8 @@ def fetchIfaceInfoFromNetworkdbAsDict(bridge, iface=None):
                     results['net-admin-configuration'].addIPv6(NetInterface.DHCP)
                 elif protov6 == 'autoconf':
                     results['net-admin-configuration'].addIPv6(NetInterface.Autoconf)
+            else:
+                raise SettingsNotAvailable("Could not find networkd.db on the file system to determine net-admin-bridge or net-admin-interface")
```

Comment:
As these exceptions may be presented to the user in some situations, I would suggest keeping the text higher level - e.g. `Could not find network configuration`  - and then logging a message with the details.

Reply:
I think you need to raise an exception in the `if not mgmt_iface:` clause as well.

Reply:
I think I've appropriately caught and handled the exception everywhere (it gets quite nested quickly)
Some places are for checking if we need to update the networking which is skipped if we don't have the information to update in the settings
Updated the exceptions to be user friendly and added logging